### PR TITLE
Added date-time to html/md report (#6084)

### LINKED ISSUE #6084: Add generation date to the HTML and MD reports
**Is your feature request related to a problem? Please describe.**
At the moment when sharing an HTML report the date the scan was performed is not on the report so it's difficult to compare reports when shared with vendors. Many companies have a scan policy of every x months. It would be really helpful to have the scan timestamp on the report. 

**Describe the solution you'd like**
Add a Timestamp of when the scan was completed to the HTML Report

**Describe alternatives you've considered**
There are no alternatives at the moment other than editing the HTML source by hand to add the timestamp.

**Additional context**
Here is a mock up of the request with exaggerated colors just so it stands out:
2020-07-21-FeatureRequestTimestampOnScanReport

**Would you like to help to implement this feature?**
The source code of ZAP is daunting. I'll leave this one to the expert programmers.
I'm guessing there is a database field with the Scan Completion Timestamp
For current scan
Read `Scan Completion Timestamp` field in database
Write Timestamp to HTML Report
(if only programming was this easy) 😸

### PR DESCRIPTION
**Pull request by StefanTobler**
Resolves #6084

### CODE REVIEW

File: zap/src/main/resources/org/zaproxy/zap/resources/xml/report.html.xsl [hunk 1]
```diff
@@ -1,11 +1,13 @@
 <?xml version="1.0" encoding="ISO-8859-1"?>
 
-<xsl:stylesheet 
+<xsl:stylesheet
```
Comment (kingthorin) [inline zap/src/main/resources/org/zaproxy/zap/resources/xml/report.html.xsl:3]: This should be reverted.

File: zap/src/main/java/org/parosproxy/paros/extension/report/ReportGenerator.java [hunk 2]
```diff
@@ -282,7 +284,7 @@ public static String stringToHtml(String inxml, StreamSource stylesource) {
         // we should really adopt something other than XSLT ;)
         return writer.toString().replace("&lt;p&gt;", "<p>").replace("&lt;/p&gt;", "</p>");
     }
-
+
```
Comment (kingthorin) [inline zap/src/main/java/org/parosproxy/paros/extension/report/ReportGenerator.java:287]: This should  be reverted.

**Comment by kingthorin** [comment]
Seems to need a spotlessApply:
`./gradlew spotlessApply`

Refresh, add, commit, push. (An amended commit would be best, though you'll have to force push in that case.)

**Comment by StefanTobler** [comment]
image

**Comment by StefanTobler** [comment]
I'll get it done now. Thank you for your patience. Super excited for my first PR on this project!

**Comment by StefanTobler** [comment]
ec2b438 should be it!

**Comment by thc202** [comment]
`ReportGenerator#stringToHtml(String, StreamSource)` and `fileToHtml` need to be updated as well, the former is used by the API.

**Comment by StefanTobler** [comment]
I’ll update it and amend the commit this evening. Thanks for letting me know!

**Comment by StefanTobler** [comment]
While adding date time to `ReportGenerator#stringToHtml(String, StreamSource)` and `fileToHtml`, I went ahead and added the xsl param to the markdown template so that the markdown file will show as follows:

```
# ZAP Scanning Report

Generated on Thu, 6 Aug 2020 23:36:36
```

**Comment by thc202** [comment]
Thanks! How would you like to be credited (e.g. name, handle)?
https://www.zaproxy.org/docs/desktop/credits/#zap-extended-team

**Comment by StefanTobler** [comment]
Stefan Tobler is fine! Thank you!

**Pull request merged**